### ISSUE TITLE
[Bug Report] Pow gradient calculation method is different with pytorch

### ISSUE THREAD

**Issue by hschoi4448** [tenstorrent-metal/tt-metal#5082]
**Describe the bug**

in Pytorch version of pow gradient computation, there are if statement for case `exponent is equal to zero`.
https://github.com/pytorch/pytorch/blob/main/torch/csrc/autograd/FunctionsManual.cpp#L466

However, the if statements does not exist in the TT version of pow gradient calculation.
https://github.com/tenstorrent-metal/tt-metal/blob/main/tt_eager/tt_dnn/op_library/backward/backward_ops.cpp#L46

so if exponent is zero, pytorch produce zero and TT produce nan as gradient.

**Expected behavior**

I expect same gradient with pytorch to avoid unncessary work finding out what's causing the nan.

**Screenshots**

**Please complete the following environment information:**

**Additional context**

**Comment by razorback3** [comment]
@chekangliang Here is a bug of current implementation of ELEMWISE_UNARY_POW.
This op is used in GPT-2.

**Comment by jliangTT** [comment]
@muthutt

**Comment by hschoi4448** [comment]
> @hschoi4448 - we already have a test case in the test suite covering exponent = 0 for the case of both backward and forward ops; I see even the case of pow(tensor,exponent=0) we have correct results.
> 
> Can you please show me a test case where this fails for you ?

Sorry, I didn't actually create a test before making the issus; I simply examined the difference in the if statement within the pow backward code and then created the issue.

Now, I'have executed the `test_backward_unary_pow.py` on the master branch and confirmed that the results are correct.


While running the test, I noticed two strange things. Should I create separate issues for them?


# 1. power(0,-1) seems not ok.

Also I created a test for zero exponent because you mentiond, `we already have a test case in the test suite covering exponent = 0, ... and have correct results`. but I think there's something wrong in `pow_bw`.

I comment out the exponent check and ran the following tests.

```Python
@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
@pytest.mark.parametrize(
    "exponent",
    [
        0.0
    ],
)
def test_bw_unary_pow(input_shapes, exponent, device):
    def my_data_gen_pt_tt(input_shapes, device, required_grad=False):
        pt_tensor = torch.zeros(input_shapes, requires_grad=required_grad).bfloat16()
        tt_tensor = (
            tt_lib.tensor.Tensor(pt_tensor, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
        )
        return pt_tensor, tt_tensor
    in_data, input_tensor = my_data_gen_pt_tt(input_shapes, device, True)
    grad_data, grad_tensor = my_data_gen_pt_tt(input_shapes, device)

    tt_output_tensor_on_device = tt_lib.tensor.unary_pow_bw(grad_tensor, input_tensor, exponent=exponent)

    in_data.retain_grad()

    pyt_y = torch.pow(in_data, exponent)

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]

    status = compare_results(tt_output_tensor_on_device, golden_tensor)
    assert status
```
This test passed, but if the operation were correct, it should have failed.

0^-1 is `inf` in pytorch.
So if `power(input, exComponent - 1, output_mem_config);` will correctly return 'inf'.
The backward output must be 'nan' if there is no exponent check.

it means the value of `pow(0, -1`) is incorrect.




# 2. not integer exponent

When I use a not integer number as the exponent, the test fails. Is this a bug, or are you planning to create `unary_power_fp_bw` separately?

```Python
@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
    ),
)
@pytest.mark.parametrize(
    "exponent",
    [
        0.5
    ],
)
def test_bw_unary_pow(input_shapes, exponent, device):
    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True)
    grad_data, grad_tensor = data_gen_pt_tt(input_shapes, device)

    tt_output_tensor_on_device = tt_lib.tensor.unary_pow_bw(grad_tensor, input_tensor, exponent=exponent)

    in_data.retain_grad()

    pyt_y = torch.pow(in_data, exponent)

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]

    status = compare_results(tt_output_tensor_on_device, golden_tensor)
    assert status

```
